packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx

```diff
@@ -111,6 +122,28 @@ export const RecordIndexOptionsDropdownContent = ({
     viewBarId: recordIndexId,
   });
 
+  const {
+    hiddenRecordGroups,
+    visibleRecordGroups,
+    viewGroupFieldMetadataItem,
+  } = useRecordGroupStates({
```

Comment:
We should rename this hook useRecordGroups because States is a convention used for hooks that return V1 component states